testutil/queryclient/queryclient.go

```diff
@@ -0,0 +1,131 @@
+package queryclient
+
+import (
+	"context"
+	"fmt"
+
+	abci "github.com/cometbft/cometbft/api/cometbft/abci/v1"
+	gogogrpc "github.com/cosmos/gogoproto/grpc"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/encoding"
+
+	"github.com/cosmos/cosmos-sdk/client/grpc/reflection"
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+)
+
+var (
+	_ gogogrpc.ClientConn = &QueryHelper{}
+	_ gogogrpc.Server     = &QueryHelper{}
+)
+
+// QueryHelper is a test utility for building a query client from a proto interface registry.
+type QueryHelper struct {
+	cdc    encoding.Codec
+	routes map[string]GRPCQueryHandler
+}
+
+func NewQueryHelper(interfaceRegistry codectypes.InterfaceRegistry) *QueryHelper {
+	// instantiate the codec
+	cdc := codec.NewProtoCodec(interfaceRegistry).GRPCCodec()
+	// Once we have an interface registry, we can register the interface
+	// registry reflection gRPC service.
+
+	qH := &QueryHelper{
+		cdc:    cdc,
+		routes: map[string]GRPCQueryHandler{},
+	}
+
+	reflection.RegisterReflectionServiceServer(qH, reflection.NewReflectionServiceServer(interfaceRegistry))
+
+	return qH
+}
+
+// Invoke implements the grpc ClientConn.Invoke method
+func (q *QueryHelper) Invoke(ctx context.Context, method string, args, reply interface{}, _ ...grpc.CallOption) error {
+	querier := q.Route(method)
+	if querier == nil {
+		return fmt.Errorf("handler not found for %s", method)
+	}
+	reqBz, err := q.cdc.Marshal(args)
+	if err != nil {
+		return err
+	}
+
+	res, err := querier(ctx, &abci.QueryRequest{Data: reqBz})
+	if err != nil {
+		return err
+	}
+
+	err = q.cdc.Unmarshal(res.Value, reply)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// NewStream implements the grpc ClientConn.NewStream method
+func (q *QueryHelper) NewStream(context.Context, *grpc.StreamDesc, string, ...grpc.CallOption) (grpc.ClientStream, error) {
+	panic("not implemented")
+}
+
+// GRPCQueryHandler defines a function type which handles ABCI Query requests
+// using gRPC
+type GRPCQueryHandler = func(ctx context.Context, req *abci.QueryRequest) (*abci.QueryResponse, error)
+
+// Route returns the GRPCQueryHandler for a given query route path or nil
+// if not found
+func (qrt *QueryHelper) Route(path string) GRPCQueryHandler {
```

Comment:
the receivers above use `q` and here it is `qrt`. i think we should just do `q` for all